qiskit/providers/ibmq/jupyter/dashboard/dashboard.py

```diff
@@ -0,0 +1,312 @@
+    """An IQX dashboard.
+    """
+    def __init__(self):
+        super().__init__()
+        self.jobs: List = []
+        self._init_subscriber()
+        self.dashboard: Accordion_with_thread = None
+        self.backend_dict: Dict = None
+        self.job_viewer: wid.VBox = None
+        self._clear_jobs_button: wid.GridBox = make_clear_button(self)
+        self._jobs_labels: wid.HBox = make_labels()
+        self.refresh_jobs_board()
+
+    def _get_backends(self):
+        from qiskit import IBMQ  # pylint: disable=import-outside-toplevel
+        ibmq_backends = {}
+        for pro in IBMQ.providers():
+            pro_name = "{hub}/{group}/{project}".format(hub=pro.credentials.hub,
+                                                        group=pro.credentials.group,
+                                                        project=pro.credentials.project)
+            for back in pro.backends():
+                if not back.configuration().simulator:
+                    if back.name() not in ibmq_backends.keys():
+                        ibmq_backends[back.name()] = [back, [pro_name]]
+                    else:
+                        ibmq_backends[back.name()][1].append(pro_name)
+
+        self.backend_dict = ibmq_backends
+
+    def refresh_jobs_board(self):
+        """Refreshes the job viewer.
+        """
+        if self.job_viewer is not None:
+            self.job_viewer.children = [self._clear_jobs_button,
+                                        self._jobs_labels] + list(reversed(self.jobs))
+
+    def refresh_device_list(self):
+        """Refresh the list of devices
+        """
+        for _wid in self.dashboard._device_list.children:
+            _wid.close()
+        self.dashboard._device_list.children = []
+        for back in self.backend_dict.values():
+            _thread = threading.Thread(target=_add_device_to_list,
+                                       args=(back, self.dashboard._device_list))
+            _thread.start()
+
+    def start_dashboard(self):
+        """Starts the job viewer
+        """
+        self.dashboard = build_dashboard_widget()
+        self.job_viewer = self.dashboard.children[0].children[1]
+        self._get_backends()
+        self.refresh_device_list()
+        self.dashboard._thread = threading.Thread(target=update_backend_info,
+                                                  args=(self.dashboard._device_list,))
+        self.dashboard._thread.do_run = True
+        self.dashboard._thread.start()
+        self.refresh_jobs_board()
+
+    def stop_dashboard(self):
+        """Stops the job viewer.
+        """
+        if self.dashboard:
+            self.dashboard._thread.do_run = False
+            self.dashboard._thread.join()
+            self.dashboard.close()
+        self.dashboard = None
+
+    def update_single_job(self, update_info: Tuple):
+        """Update a single job instance
+
+        Args:
+            update_info (tuple): Updated job info.
```

Comment:
Can you document what the expected elements are in this tuple? Or better yet, use a helper class or individual parameters so one doesn't need to guess what's in each index location.

Reply:
done.